Tutorials/Containers.md

```diff
@@ -0,0 +1,598 @@
+This portion of the tutorial assumes that the user has sufficient 
+admin privileges to install/configure singularity and/or has obtained 
+permission from the appropriate administration staff to utilize `fakeroot`.  
+In this tutorial, we will provide two main paths for building and utilizing 
+Singularity containers for SST.  The first method provides users the ability 
+to build and utilize basic SST containers without support for networking 
+across containers.  These "monolithic" containers can be utilized for initial 
+testing and developing of external modules.  The second method provides users 
+the ability to build and utilize advanced SST containers that support networking 
+and communication (MPI) across individual container instances.  Thus, the 
+advanced (parallel) method provides much greater scalability of simulations.  
+
+### Singularity Installation <a name="singularityinstallation"></a>
+
+In order to install Singularity on your respective system, utilize 
+the instructions provided by the Singularity installation guides.  For this 
+tutorial, we assume the use of Singularity version 3.8. Pay special attention 
+to the guidelines associated with installing and utilizing the `Go` infrastructure.
+
+The installation guide for Singularity 3.8 can be found here: 
+
+[Singularity Installation](https://sylabs.io/guides/3.8/user-guide/quick_start.html#quick-installation-steps)
+
+### Building Basic Singularity Containers <a name="buildingbasicsingularitycontainers"></a>
+
+#### Pre-Built Formulas <a name="basicprebuildformulas"></a>
+
+We *HIGHLY* suggest that users utilize the pre-built singularity container 
+formulas.  These formulas contain basic installations of SST-Core and/or SST-Elements
+that can be utilized and/or modified to serve as the basis for a functional 
+SST environment.  The pre-built SST container repository is found below.  
+
+[SST Container Repository](https://github.com/tactcomplabs/sst-containers)
+
+This repository contains two types of singularity containers.  The first includes the SST-Core
+package and the second contains the SST-Core and SST-Elements packages.  The naming
+convention for the singularity `def` files is as follows:
+
+* sstcore-VERSION-OS.def : SST-Core only
+* sstpackage-VERSION-OS.def : SST-Core + SST-Elements
+
+The version numbers supported are listed as follows:
+* 11.0.0 : SST 11.0.0 Release
+* MASTER : SST github master branches
+
+The supported operating systems are listed as follows:
+* ubuntu-18.04
+* ubuntu-20.04
+* centos7
+* centos8
+
+Prior to building the provided SST containers, you must first bootstrap the 
+environment.  The repository contains a `bootstrap.sh` script that can be 
+utilized to initialize the environment, find available packages and download 
+the appropriate SST source packages.  Bootstrapping the environment can be performed 
+as follows:
+
+```
+$> git clone https://github.com/tactcomplabs/sst-containers.git
```

Comment:
if it's possible to not depend on external repos, that would be preferable.

Reply:
We can do one of several things:

- merge the sst-containers repo over to the official SST organization
- setup a public container registry for which public users can pull pre-built container images

There are likely other options, so I'm not opposed to exploring further

Reply:
@gvoskuilen -- what do you think of having "sst-containers" as part of github.com/sstsimulator ?

Reply:
We're definitely good with that.  I would prefer to delay that until we have the CentOS{7,8} containers functional post EOL before we migrate things over.